hcl2template/types.build.hcp_packer_registry_test.go

```diff
@@ -287,6 +332,615 @@ func Test_ParseHCPPackerRegistryBlock(t *testing.T) {
+				Sources: map[SourceRef]SourceBlock{
+					refNull: {
+						Type: "null",
+						Name: "test",
+						block: &hcl.Block{
+							Type: "source",
+						},
+					},
+				},
+				Builds: Builds{
+					{
+						Name: "bucket-slug",
+						Sources: []SourceUseBlock{
+							{
+								SourceRef: refNull,
+							},
+						},
+					},
+				},
+			},
+			true, true,
+			nil,
+			false,
+			&getHCPPackerRegistry{
+				wantBlock:        &HCPPackerRegistryBlock{},
+				wantDiag:         false,
+				wantDiagHasError: false,
+			},
+		},
+		{"bucket name too short",
+			defaultParser,
+			parseTestArgs{"testdata/hcp_par/top-level-short-bucket.pkr.hcl", nil, nil},
+			&PackerConfig{
+				CorePackerVersionString: lockedVersion,
+				Basedir:                 filepath.Join("testdata", "hcp_par"),
+				Sources: map[SourceRef]SourceBlock{
+					refNull: {
+						Type: "null",
+						Name: "test",
+						block: &hcl.Block{
+							Type: "source",
+						},
+					},
+				},
+				Builds: Builds{
+					{
+						Name: "bucket-slug",
+						Sources: []SourceUseBlock{
+							{
+								SourceRef: refNull,
+							},
+						},
+					},
+				},
+			},
+			true, true,
+			nil,
+			false,
+			&getHCPPackerRegistry{
+				wantBlock:        &HCPPackerRegistryBlock{},
+				wantDiag:         false,
+				wantDiagHasError: false,
+			},
+		},
+		{"bucket name too long",
+			defaultParser,
+			parseTestArgs{"testdata/hcp_par/top-level-long-bucket.pkr.hcl", nil, nil},
+			&PackerConfig{
+				CorePackerVersionString: lockedVersion,
+				Basedir:                 filepath.Join("testdata", "hcp_par"),
+				Sources: map[SourceRef]SourceBlock{
+					refNull: {
+						Type: "null",
+						Name: "test",
+						block: &hcl.Block{
+							Type: "source",
+						},
+					},
+				},
+				Builds: Builds{
+					{
+						Name: "bucket-slug",
+						Sources: []SourceUseBlock{
+							{
+								SourceRef: refNull,
+							},
+						},
+					},
+				},
+			},
+			true, true,
+			nil,
+			false,
+			&getHCPPackerRegistry{
+				wantBlock:        &HCPPackerRegistryBlock{},
+				wantDiag:         false,
+				wantDiagHasError: false,
+			},
+		},
+		{"bucket name invalid chars",
+			defaultParser,
+			parseTestArgs{"testdata/hcp_par/top-level-invalid-bucket.pkr.hcl", nil, nil},
+			&PackerConfig{
+				CorePackerVersionString: lockedVersion,
+				Basedir:                 filepath.Join("testdata", "hcp_par"),
+				Sources: map[SourceRef]SourceBlock{
+					refNull: {
+						Type: "null",
+						Name: "test",
+						block: &hcl.Block{
+							Type: "source",
+						},
+					},
+				},
+				Builds: Builds{
+					{
+						Name: "bucket-slug",
+						Sources: []SourceUseBlock{
+							{
+								SourceRef: refNull,
+							},
+						},
+					},
+				},
+			},
+			true, true,
+			nil,
+			false,
+			&getHCPPackerRegistry{
+				wantBlock:        &HCPPackerRegistryBlock{},
+				wantDiag:         false,
+				wantDiagHasError: false,
+			},
+		},
+		{"bucket name OK",
```

Comment:
I can see a reason why we'd want to test that the top-level block logic matches the build-block one, but since it's the same code performing the checks, I'm wondering if we need to.
At this point, I think we can have only top-level tests since this is the new normal, and we should add tests to make sure the right block/warning/errors are produced, but besides that, it should be essentially enough, right?